x-pack/elastic-agent/pkg/core/server/server_test.go

```diff
@@ -757,7 +757,7 @@ func (*SleepAction) Name() string {
 	return "sleep"
 }
 
-func (*SleepAction) Execute(request map[string]interface{}) (map[string]interface{}, error) {
+func (*SleepAction) Execute(ctx context.Context, request map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
Why is the context needed, and why is the sleep not cancellable. For cancellation consider [`timed.Wait`](https://github.com/elastic/go-concert/blob/master/timed/timed.go#L39).

Reply:
it implements the Action interface
https://github.com/elastic/elastic-agent-client/blob/7dd05ee2b5a5d7c6800117d7a49375d87294a716/pkg/client/client.go#L60

Reply:
Does the `Action` interface use `context.Context` for cancellation or is it require to pass values via `Values`?

See line 769: The sleep is not cancellable and should be changed to `err := timed.Wait(ctx, time.Duration(sleep))` in order to make the action cancellable.

Reply:
ahh, misunderstood the previous comment, will update SleepAction implementation